### PR TITLE
feat(app): Use the v8 backend

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Issue

Closes: AVO-6

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
I don't plan on merging this anytime soon as I want all of the other contrib changes deployed before we merge and deploy this to take your time.

There is a lot of changes in here so don't hesitate to adress anything you feel is wrong or strange.

**Comment by VIKTORVAV99** [comment]
Added JSDoc comments to the new types so there is a explanation of the value like this:
image

File: web/src/features/exchanges/ExchangeTooltip.tsx
Comment (VIKTORVAV99) [inline web/src/features/exchanges/ExchangeTooltip.tsx:21]: This will never be triggered as the ExchangeArrow function where this is called from has already exited early. I'll do a bigger revamp, code cleanup and deduplication of these tooltips in a different PR.

**Comment by madsnedergaard** [comment]
Let me know when this is ready for review :)

**Comment by VIKTORVAV99** [comment]
> Let me know when this is ready for review :)

The frontend is good to go but I think I found a bug in the backend with the estimation filtering I need to figure out first before we deploy this to all users.

But if you want to get a head start on the review feel free to do so!

**Comment by VIKTORVAV99** [comment]
I'll take care of the conflict but otherwise this PR is now good to go!

File: web/src/features/exchanges/ExchangeTooltip.tsx
Comment (madsnedergaard) [inline web/src/features/exchanges/ExchangeTooltip.tsx:20]: I'm sorry for pointing this out quite late, but I think we should try to do the remapping as early as possible instead of using the shortened version all the way down.

In this exchange we could make the remapping in `useExchangeArrowsData` so it's passed down :)

Comment (VIKTORVAV99) [reply web/src/features/exchanges/ExchangeTooltip.tsx:20]: Will take care of it later today 🙂

Comment (madsnedergaard) [reply web/src/features/exchanges/ExchangeTooltip.tsx:20]: Thanks, I know it's quite late to bring this up 😅

File: web/src/features/panels/ranking-panel/getRankingPanelData.ts
Comment (madsnedergaard) [inline web/src/features/panels/ranking-panel/getRankingPanelData.ts:36]: Is this supposed to be `grid`?

Comment (VIKTORVAV99) [reply web/src/features/panels/ranking-panel/getRankingPanelData.ts:36]: I think so, as in grid state.

But I'll double check 🙂

File: web/src/features/map/MapTooltip.tsx
Comment (madsnedergaard) [inline web/src/features/map/MapTooltip.tsx:74]: I think this idea of a property that can be both a number and a boolean is going to cause problems down the road, but we can reconsider this at a later point :)

File: web/src/features/panels/ranking-panel/getRankingPanelData.ts
Comment (madsnedergaard) [inline web/src/features/panels/ranking-panel/getRankingPanelData.ts:36]: `gird` should probably be grid here too, right?

File: web/src/features/panels/zone/ZoneHeaderGauges.tsx
Comment (madsnedergaard) [inline web/src/features/panels/zone/ZoneHeaderGauges.tsx:49]: couldn't we just pass the whole "selectedData" variable to the util functions here instead of reconstructing the object?

**Review by madsnedergaard — COMMENTED** [review]
Thanks for making the changes, it's definitely much easier to understand and use now 😄 
A few additional comments

File: web/src/features/map/MapTooltip.tsx
Comment (VIKTORVAV99) [reply web/src/features/map/MapTooltip.tsx:74]: Possibly but considering what we learned about the estimations recently I think that maybe it should just be a number for all time aggregations.

File: web/src/features/panels/ranking-panel/getRankingPanelData.ts
Comment (VIKTORVAV99) [reply web/src/features/panels/ranking-panel/getRankingPanelData.ts:36]: That's what I get for trying to spell in the morning 😅

I'll take care of it...

File: web/src/features/panels/zone/ZoneHeaderGauges.tsx
Comment (VIKTORVAV99) [reply web/src/features/panels/zone/ZoneHeaderGauges.tsx:49]: Possibly, and might be a better solution. I'll take a look!

Comment (VIKTORVAV99) [reply web/src/features/panels/zone/ZoneHeaderGauges.tsx:49]: So this is because this is using the zone data and not the state data, when the state data is used it's passed like this:
```js
  const intensity = getCarbonIntensity(zoneData, isConsumption);
```

So if we want to use the same function for both one of them need reconstruct the object in one place or the other and I think it's better the zone details should do it  than the state related things as they are used more and need to be faster.

I'm considering changing the usage of the carbon intensity from the details file to the state files though as those are already loaded. That means they will be used the same way. But that is a separate PR for sure.

Comment (VIKTORVAV99) [reply web/src/features/panels/zone/ZoneHeaderGauges.tsx:49]: It would be possible to make the function handle both kinds of data and have internal logic on which to use but unsure how beneficial that would be over constructing the object. So I'll leave that up to you to decide then I'll implement it if we decide to do so.

Comment (madsnedergaard) [reply web/src/features/panels/zone/ZoneHeaderGauges.tsx:49]: Ah okay, I see - let's keep it like this then :)

**Comment by madsnedergaard** [comment]
I just hit the same "state and details not in sync" problem that we've had before, where visiting at 21:55 meant the numbers were out of sync due to caching. This became more clear while testing this PR as no users are using `v8` yet. I suggest we pick up this problem soon-ish after the app-backend caching project is done :)

Screenshot 2024-04-11 at 22 00 03
Screenshot 2024-04-11 at 21 59 49

Don't think it's blocking for this PR though, just thought I'd share it since we talked about it earlier today.

**Review by madsnedergaard — APPROVED** [review]

**Comment by VIKTORVAV99** [comment]
Will create a new release tomorrow morning which includes this and monitor it during the day for any issues.

**Comment by VIKTORVAV99** [comment]
Thanks for the reviews!

**Comment by tonypls** [comment]
I know I'm late to the party here but it feels like the right place to discuss.

Would it be beneficial to convert all shortened keys into human-readable ones at the API query level? Revisiting the codebase and having to refer to an interface to decode a variable adds another layer of complexity.

Consider the difference in cognitive load between `const { e, o } = zoneData;` and `const { isEstimated, hasOutage } = zoneData;`. Making variable names human-readable from the API layer down could enhance the developer experience by improving readability. Although there might be a slight increase in bundle size, variable names are generally minified, so the impact should be negligible. 

What are your thoughts on prioritizing readability in this context?

**Comment by VIKTORVAV99** [comment]
> I know I'm late to the party here but it feels like the right place to discuss.
> 
> 
> 
> Would it be beneficial to convert all shortened keys into human-readable ones at the API query level? Revisiting the codebase and having to refer to an interface to decode a variable adds another layer of complexity.
> 
> 
> 
> Consider the difference in cognitive load between `const { e, o } = zoneData;` and `const { isEstimated, hasOutage } = zoneData;`. Making variable names human-readable from the API layer down could enhance the developer experience by improving readability. Although there might be a slight increase in bundle size, variable names are generally minified, so the impact should be negligible. 
> 
> 
> 
> What are your thoughts on prioritizing readability in this context?
> 
> 

I would be happy to improve the Developer Experience of this, especially if we end up changing the details data as well as that's even more complex. But I'm not sure how much of a priority it should be?

Bundle size issue of that in particular would be almost non existent so I don't think that should be a concern. A bigger concern is the runtime cost of converting all the keys IMO.

I can take a look at it during the slack week and then we can make a more informed decision on what we want to do if you wish?

**Comment by tonypls** [comment]
> > I know I'm late to the party here but it feels like the right place to discuss.
> > Would it be beneficial to convert all shortened keys into human-readable ones at the API query level? Revisiting the codebase and having to refer to an interface to decode a variable adds another layer of complexity.
> > Consider the difference in cognitive load between `const { e, o } = zoneData;` and `const { isEstimated, hasOutage } = zoneData;`. Making variable names human-readable from the API layer down could enhance the developer experience by improving readability. Although there might be a slight increase in bundle size, variable names are generally minified, so the impact should be negligible.
> > What are your thoughts on prioritizing readability in this context?
> 
> I would be happy to improve the Developer Experience of this, especially if we end up changing the details data as well as that's even more complex. But I'm not sure how much of a priority it should be?
> 
> Bundle size issue of that in particular would be almost non existent so I don't think that should be a concern. A bigger concern is the runtime cost of converting all the keys IMO.
> 
> I can take a look at it during the slack week and then we can make a more informed decision on what we want to do if you wish?

I do find it slows me down significantly when navigating the code base, as an example I often highlight and jump to variable names but in this case it's no longer a possibility: 
image

Ideally, we should prioritize enhancements to developer experience that restore the usability we previously enjoyed. I have created a ticket to address this [here](https://linear.app/electricitymaps/issue/AVO-190/map-v8-api-to-human-readable-variable-names) happy to pick it up myself during slack week too if you're busy.

**Comment by VIKTORVAV99** [comment]
> > > I know I'm late to the party here but it feels like the right place to discuss.
> > > Would it be beneficial to convert all shortened keys into human-readable ones at the API query level? Revisiting the codebase and having to refer to an interface to decode a variable adds another layer of complexity.
> > > Consider the difference in cognitive load between `const { e, o } = zoneData;` and `const { isEstimated, hasOutage } = zoneData;`. Making variable names human-readable from the API layer down could enhance the developer experience by improving readability. Although there might be a slight increase in bundle size, variable names are generally minified, so the impact should be negligible.
> > > What are your thoughts on prioritizing readability in this context?
> > 
> > 
> > I would be happy to improve the Developer Experience of this, especially if we end up changing the details data as well as that's even more complex. But I'm not sure how much of a priority it should be?
> > Bundle size issue of that in particular would be almost non existent so I don't think that should be a concern. A bigger concern is the runtime cost of converting all the keys IMO.
> > I can take a look at it during the slack week and then we can make a more informed decision on what we want to do if you wish?
> 
> I do find it slows me down significantly when navigating the code base, as an example I often highlight and jump to variable names but in this case it's no longer a possibility: image
> 
> Ideally, we should prioritize enhancements to developer experience that restore the usability we previously enjoyed. I have created a ticket to address this [here](https://linear.app/electricitymaps/issue/AVO-190/map-v8-api-to-human-readable-variable-names) happy to pick it up myself during slack week too if you're busy.

Makes sense. Although if you limited that to match exactly you'd find it pretty fast 😉

Personally I usually use the reference navigator which shows you exactly where an object is used:
image

**Comment by madsnedergaard** [comment]
I missed that we didn't update it everywhere, we should indeed have the full, readable names wherever possible IMO :)

**Pull request merged**
